openpgp/packet/public_key.go

```diff
@@ -371,15 +370,46 @@ func (pk *PublicKey) parseECDH(r io.Reader) (err error) {
 		return errors.UnsupportedError("unsupported ECDH KDF cipher: " + strconv.Itoa(int(pk.kdf.Bytes()[2])))
 	}
 
+	kdf := ecdh.KDF{
```

Comment:
```suggestion
	kdf := &ecdh.KDF{
```
I kinda feel like this should be a reference, in order not to make a copy when you write it to the `PublicKey`. But I guess it would technically be a backwards-incompatible change, if you have to change the type of `ecdh.PublicKey.KDF`. But maybe it's specific enough that we can get away with changing it.

But ok, if it turns out that you also have to change tests etc to change this, then just ignore this comment.